test/broadcast.jl

```diff
@@ -417,8 +417,8 @@ StrangeType18623(x,y) = (x,y)
 let
     f(A, n) = broadcast(x -> +(x, n), A)
     @test @inferred(f([1.0], 1)) == [2.0]
-    g() = (a = 1; Broadcast.combine_eltypes(x -> x + a, (1.0,)))
-    @test @inferred(g()) === Float64
+    g() = (a = 1; x -> x + a)
+    @test @inferred(broadcast(g(), 1.0)) === 2.0
```

Comment:
@pabloferz @Sacha0 Since you worked on these tests (this one and the one below), could you confirm that the new ones covers the same use case as the old ones? That wasn't completely clear to me.

Reply:
Regrettably sufficient time has elapsed since I looked at these tests and such that I no longer have much memory of them. Sorry Milan! :)